core/services/keystore/keys/ocr2key/key_bundle.go

```diff
@@ -106,7 +106,7 @@ func (raw Raw) Key() (kb KeyBundle) {
 	case chaintype.StarkNet:
 		kb = newKeyBundle(new(starknet.OCR2Key))
 	default:
-		panic(chaintype.NewErrInvalidChainType(temp.ChainType))
+		return
```

Comment:
I wonder if we should return an error... :thinking:  There is another panic on Unmarshal that also seems extreme, and it would be nice to log something up the stack in both cases.

Do the other `Key()` methods `panic` too?

Reply:
Hmmm yeah, we might have to find a non-destructive, non-panicing solution here, but I'm not very familiar with this part of the code.
@samsondav @RyanRHall @connorwstein  WDYT?

Reply:
Yeah this is a valid point. We may need to do something a bit more clever here, archiving keys and exporting them somehow?

Reply:
Added a small "legacy key" system inside the keyring that keeps track of unsupported keys. Keys are loaded into the system when decrypt is called and unloaded on encrypt.